delfin/drivers/dell_emc/vnx/vnx_block/component_handler.py

```diff
@@ -565,3 +572,340 @@ def list_storage_hosts(self, storage_id):
+            if not resources_map.get(resource_obj) \
+                    or not resources_type_map.get(resource_obj):
+                continue
+            resources_type = resources_type_map.get(resource_obj)
+            labels = {
+                'storage_id': storage_id,
+                'resource_type': resources_type,
+                'resource_id': resources_map.get(resource_obj),
+                'type': 'RAW',
+                'unit': ''
+            }
+            if not performance_lines_map.get(resource_obj):
+                continue
+            metric_model_list = self._get_metric_model(
+                resource_metrics.get(resources_type), labels,
+                performance_lines_map.get(resource_obj),
+                consts.RESOURCES_TYPE_TO_METRIC_CAP.get(
+                    resources_type), resources_type)
+            if metric_model_list:
+                metrics.extend(metric_model_list)
+        return metrics
+
+    def _get__archive_file(self, start_time, end_time):
+        archive_file_list = []
+        archives = self.navi_handler.get_archives()
+        tools = Tools()
+        for archive_info in (archives or []):
+            collection_timestamp = tools.time_str_to_timestamp(
+                archive_info.get('collection_time'), consts.TIME_PATTERN)
+            if collection_timestamp > start_time:
+                archive_file_list.append(archive_info.get('archive_name'))
+            if collection_timestamp > end_time:
+                break
+        return archive_file_list
+
+    def _get_metric_model(self, metric_list, labels, metric_values, obj_cap,
+                          resources_type):
+        metric_model_list = []
+        if not metric_values:
+            return metric_model_list
+        tools = Tools()
+        for metric_name in (metric_list or []):
+            values = {}
+            obj_labels = copy.deepcopy(labels)
+            obj_labels['unit'] = obj_cap.get(metric_name).get('unit')
+            for metric_value in metric_values:
+                metric_value_infos = metric_value
+                if not consts.METRIC_MAP.get(resources_type):
+                    continue
+                if not consts.METRIC_MAP.get(resources_type).get(metric_name):
+                    continue
+                value = metric_value_infos[
+                    consts.METRIC_MAP.get(resources_type).get(metric_name)]
+                if not value:
+                    value = '0'
+                collection_timestamp = tools.time_str_to_timestamp(
+                    metric_value_infos[1], consts.TIME_PATTERN)
+                collection_time_str = tools.timestamp_to_time_str(
+                    collection_timestamp, consts.COLLECTION_TIME_PATTERN)
+                collection_timestamp = tools.time_str_to_timestamp(
+                    collection_time_str, consts.COLLECTION_TIME_PATTERN)
+                if "iops" in metric_name.lower():
```

Comment:
It's recommended to use `==` , not `in` .

Reply:
Modified